src/store/juju/slice.ts

```diff
@@ -108,6 +123,31 @@ const slice = createSlice({
     clearControllerData: (state) => {
       state.controllers = {};
     },
+    fetchCrossModelQuery: (
+      state,
+      action: PayloadAction<CrossModelQueryRequest & WsControllerURLParam>
+    ) => {
+      state.crossModelQuery.loading = true;
+    },
+    updateCrossModelQuery: (
+      state,
+      { payload }: PayloadAction<CrossModelQueryFullResponse>
+    ) => {
+      state.crossModelQuery.results = isCrossModelQueryResponse(payload)
+        ? payload.results
+        : null;
+      state.crossModelQuery.errors = isCrossModelQueryResponse(payload)
+        ? payload.errors
+        : payload;
```

Comment:
It might be worth leaving a comment here to explain that a string response from the API is an error (similar to the comment you left on the type in jimm-facade.ts).